### ISSUE TITLE
prediction on Imported model causes exception

### ISSUE THREAD

**Issue by AlexanderZender** [keras-team/autokeras#1589]
### Bug Description
Whenever I try to use an exported model the following exception occurs on calling the predict() function:

Failed to convert a NumPy array to a Tensor (Unsupported object type int).

Used dataset is the Titanic dataset from Kaggle.

Any ideas on what I might be missing here?

Thank you

### Bug Reproduction
Use the Titanic Dataset to generate a new model, export said model to your disk.
In a separate project import the model using the example code from Autokeras.
Read the Titanic Dataset and try to make a prediction

### Expected Behavior
No exception and make a prediction on the test dataset.

### Setup Details
Include the details about the versions of:
 - OS type and version: Windows
 - Python: 3.7
 - autokeras: 1.0.14
 - keras-tuner: 1.0.2
 - scikit-learn: 0.24.2
 - numpy: 1.19.5
 - pandas: 1.2.4
 - tensorflow: 2.5

### Additional context

Program to perform AutoML process:

`training_data_path = os.path.join(request.dataset_path, request.dataset_filename)
        
        training_file = training_data_path
        clf = ak.StructuredDataClassifier(overwrite=True, max_trials=3, seed=42)
        clf.fit(training_file, "survived", epochs=10)

        model = clf.export_model()
        model.save("model_autokeras", save_format="tf")`

Program to import model and make prediction:
`x_to_predict = pd.read_csv("C:/Users/********/Desktop/titanic-dataset/titanic_test.csv", quotechar='"', skipinitialspace=True)

loaded_model = load_model("model_autokeras", custom_objects=ak.CUSTOM_OBJECTS)
	x_to_predict_numpy = x_to_predict.to_numpy()
	predicted_y = loaded_model.predict(tf.convert_to_tensor(x_to_predict_numpy), -1)
	print(predicted_y)`

**Comment by AlexanderZender** [comment]
Fixed it by converting my data to unicode before calling predict():

`predicted_y = loaded_model.predict(x_to_predict_numpy.astype(np.unicode))`

**Comment by AlexanderZender** [comment]
I made some more testing, and it must be related to the import into Tensorflow.
While using the same dataset inside the AutoML, the prediction is binary (0. or 1.):
![image](https://user-images.githubusercontent.com/79153884/122418750-b8ee6600-cf8a-11eb-82df-3b5c25beaf85.png)


But inside Keras it is a Regression(?) (1.457812154 or 8.147856324)

![image](https://user-images.githubusercontent.com/79153884/122418793-c3106480-cf8a-11eb-8905-4f2c41a4707b.png)


`x_to_predict = pd.read_csv("C:/Users/hda10126/Desktop/titanic-dataset/titanic_test.csv", quotechar='"', skipinitialspace=True)
x_to_predict_numpy = x_to_predict.to_numpy()
clf = ak.StructuredDataClassifier(overwrite=True, max_trials=3, seed=42)
clf.fit("C:/Users/hda10126/Desktop/titanic-dataset/titanic_train.csv", "survived", epochs=10)
result = clf.predict(x_to_predict_numpy.astype(np.unicode))
print(result)
model = clf.export_model()
model.save("model_autokeras", save_format="tf")
loaded_model = load_model("model_autokeras", custom_objects=ak.CUSTOM_OBJECTS)
predicted_y = loaded_model.predict(x_to_predict_numpy.astype(np.unicode))
print(predicted_y)`

**Comment by AlexanderZender** [comment]
Ok, my problem was that the Sigmoid postprocessing wasnt included inside the export.

Perhaps its an obvious issue, but for anyone who encounters the same situation, you need to post-process the prediction:

`predicted_y[predicted_y < 0.5] = 0 
predicted_y[predicted_y > 0.5] = 1`
